crates/nri-resctrl-plugin/src/lib.rs

```diff
@@ -20,6 +20,53 @@ use nri::events_mask::EventMask;
 
 use resctrl::{Config as ResctrlConfig, FsProvider, RealFs, Resctrl};
 
+/// Source of PIDs for a container based on cgroup metadata.
+pub trait CgroupPidSource: Send + Sync {
+    fn pids_for_container(&self, c: &nri::api::Container) -> Vec<i32>;
+}
+
+pub struct RealCgroupPidSource;
```

Comment:
1. let's move the pid source to a module, this is a large file as it is.
2. use the cgroups_fs crate to enumerate PIDs